src/CommandingOfficers/Modifiers/CODamageModifier.java

```diff
@@ -20,36 +20,12 @@
   public CODamageModifier(int firepowerChange)
   {
     attackModifier = firepowerChange;
-    modelsToModify = new ArrayList<UnitModel>();
-  }
-
-  /**
-   * Add a UnitModel type for this modifier to affect. If no models are added
-   * this way, then by default apply will affect all of the Commander's UnitModels.
-   * @param model
-   */
-  public void addApplicableUnitModel(UnitModel model)
-  {
-    if( model != null )
-    {
-      modelsToModify.add(model);
-    }
-    else
-    {
-      System.out.println("Attempting to add null model to COMovementModifier!");
-      throw new NullPointerException(); // Make sure this oversight doesn't go unnoticed.
-    }
   }
 
   @Override
-  public void apply(Commander commander)
+  public void applyChanges(Commander commander, ArrayList<UnitModel> models)
```

Comment:
`applyChanges` and `revertChanges` are `protected` in `GenericCOModifier`. I'm assuming you meant for them to be protected here too.

As an aside, I would almost rather have the public-facing method be named `applyChanges`, and make `apply` the internal-only one. Small change, but it might make it more clear just that it's not attaching the modifier to the Commander itself, and so the changes won't be undone automatically.